lib/portage/tests/update/test_update_dbentry.py

```diff
@@ -520,3 +527,175 @@ def testUpdateDbentryDbapiTestCaseWithSignature(self):
 
                 finally:
                     playground.cleanup()
+
+    def testLocalFilePkgSyncUpdate(self):
+        """
+        Check handling of local file:// sync-uri and unnecessary BUILD_ID
+        increments (bug #921208).
+        """
+        debug = False
+
+        ebuilds = {
+            "dev-libs/A-1::local": {
+                "EAPI": "7",
+                "SLOT": "0",
+            },
+        }
+
+        playground = ResolverPlayground(ebuilds=ebuilds, debug=debug)
+        settings = playground.settings
+        eprefix = settings["EPREFIX"]
+        eroot = settings["EROOT"]
+        trees = playground.trees
+        bindb = trees[eroot]["bintree"].dbapi
+        var_cache_edb = os.path.join(eprefix, "var", "cache", "edb")
+        user_config_dir = os.path.join(eprefix, USER_CONFIG_PATH)
+
+        portage_python = portage._python_interpreter
+        emerge_cmd = (
+            portage_python,
+            "-b",
+            "-Wd",
+            os.path.join(str(self.bindir), "emerge"),
+        )
+
+        tmppkgdir = tempfile.TemporaryDirectory()
+        tmppkgdir_suffix = os.path.join(tmppkgdir.name, "binpkg")
+
+        test_commands = (
+            # Create a trivial binpkg first.
+            emerge_cmd
+            + (
+                "--oneshot",
+                "--verbose",
+                "--buildpkg",
+                "dev-libs/A",
+            ),
+            # Copy to a new PKGDIR which we'll use as PORTAGE_BINHOST then delete the old PKGDIR
+            (
+                (
+                    lambda: shutil.copytree(bindb.bintree.pkgdir, tmppkgdir_suffix)
+                    or True,
+                )
+            ),
+            (
+                (
+                    lambda: os.unlink(
+                        os.path.join(
+                            bindb.bintree.pkgdir, "dev-libs", "A", "A-1-1.gpkg.tar"
+                        )
+                    )
+                    or True,
+                )
+            ),
+            # This should succeed if we've correctly saved it as A-1-1.gpkg.tar, not
+            # A-1-2.gpkg.tar, and then also try to unpack the right filename.
+            emerge_cmd
```

Comment:
We can make this more finely-grained so we store the exit code, then check what the downloaded binpkg filename is, and then fail earlier on that, or fail if the earlier-stored exit code was bad too. That way, we get two distinct failures with and without https://github.com/gentoo/portage/pull/1226.